src/elasticsearch/addl_index_transformations/portal/add_partonomy.py

```diff
@@ -13,23 +37,31 @@ def add_partonomy(doc):
     ... }
     >>> from json import dumps
     >>> doc = {
+    ...     'origin_sample': {
+    ...         'organ': 'LI'
+    ...     },
     ...     'rui_location': dumps(rui_location)
     ... }
     >>> add_partonomy(doc)
     >>> del doc['rui_location']
+    >>> del doc['origin_sample']
     >>> doc
     {'anatomy_0': 'body', 'anatomy_1': 'large intestine', 'anatomy_2': 'transverse colon'}
 
     '''
-    if 'rui_location' not in doc:
-        return
-    rui_location = loads(doc['rui_location'])
+    annotations = []
+
+    organ_iri = get_organ_iri(doc)
+    if organ_iri:
+        annotations.append(organ_iri)
+
+    if 'rui_location' in doc:
+        rui_location = loads(doc['rui_location'])
 
-    if 'ccf_annotations' not in rui_location:
-        return
-    ccf_annotations = rui_location['ccf_annotations']
+        if 'ccf_annotations' in rui_location:
+            annotations += rui_location['ccf_annotations']
```

Comment:
```suggestion
        annotations += rui_location.get('ccf_annotations', [])
```

Reply:
Oops, commit suggestion didn't work right. Anyway, I'll let you take over from here.